src/lib/replication/repl_dev/raft_state_machine.cpp

```diff
@@ -320,8 +328,16 @@ int RaftStateMachine::read_logical_snp_obj(nuraft::snapshot& s, void*& user_ctx,
 
 void RaftStateMachine::save_logical_snp_obj(nuraft::snapshot& s, ulong& obj_id, nuraft::buffer& data, bool is_first_obj,
                                             bool is_last_obj) {
+    if ((obj_id & snp_obj_id_type_mask) == 0) {
```

Comment:
Can we move `(obj_id & snp_obj_id_type_mask) == 0` to a function, e.g. `is_hs_msg(obj_id)`